api4/apitestlib.go

```diff
@@ -560,57 +561,34 @@ func GenerateTestId() string {
 func CheckUserSanitization(t *testing.T, user *model.User) {
 	t.Helper()
 
-	if user.Password != "" {
-		t.Fatal("password wasn't blank")
-	}
-
-	if user.AuthData != nil && *user.AuthData != "" {
-		t.Fatal("auth data wasn't blank")
-	}
-
-	if user.MfaSecret != "" {
-		t.Fatal("mfa secret wasn't blank")
-	}
+	require.Equal(t, "", user.Password, "password wasn't blank")
+	require.Empty(t, user.AuthData, "auth data wasn't blank")
+	require.Equal(t, "", user.MfaSecret, "mfa secret wasn't blank")
 }
 
 func CheckEtag(t *testing.T, data interface{}, resp *model.Response) {
 	t.Helper()
 
-	if !reflect.ValueOf(data).IsNil() {
-		t.Fatal("etag data was not nil")
-	}
-
-	if resp.StatusCode != http.StatusNotModified {
-		t.Log("actual: " + strconv.Itoa(resp.StatusCode))
-		t.Log("expected: " + strconv.Itoa(http.StatusNotModified))
-		t.Fatal("wrong status code for etag")
-	}
+	assert.Empty(t, data)
```

Comment:
I'm a little worried about using `assert` in these checks. Normally the rule would be to use `assert` if the test can continue and `require` if it must stop, but being these helpers as they are, any test could be relying on them to stop the execution when used.

I think that to preserve the previous behaviour, is better to replace the instances of `t.Fatal` checks with `require` instead of `assert` for the specific case of this file.

Reply:
I notice that some of the functions are helper. But I see the params name is a represent Response which means it's the final state and uses `assert` instead.

Should I change all to require or only the line you mentioned?

Reply:
It is true that the helpers receive a `Response` object, but we have test cases where we use multiple requests in sequence that depend on each other, so having a response might not be the final stage of the test.

I would change all the instances of `assert` with `require`s, for consistency.